### ISSUE TITLE
[2017.2.0-rc2] Evaluation of True/False Values is Always True

### ISSUE THREAD

**Issue by ColonelPanicks** [alces-software/metalware#264]
The evaluation of true/false values inside ERB tags seem to evaluate to true when used within an if statement, take the following script:
```
<%= config.repoconfig.is_server %>
<%= config.repoconfig.is_server.class %>
<% if config.repoconfig.is_server -%>
This is true
<% else -%>
This is false
<% end -%>
```
When this is rendered, the following output is seen:
```
[root@controller ~]# metal render /tmp/test2 local
false
FalseClass
This is true
```

**Comment by bobwhitelock** [comment]
I think this is sort of a regression since `2017.1`, though the whole way templating works has changed since then. We used to explicitly handle converting `'false'` to `false` and `'true'` to `true` with the old templating system (https://github.com/alces-software/metalware/blob/2017.1.0/src/templating/repo_config_parser.rb#L136,L149) though in a not very clever way - possibly we need something similar in the new system?

Could the issue be that values are being converted to strings as the final step before being rendered? If this was the case does arithmetic with config values (e.g. `<%= config.repoconfig.some_number + 2 %>`) also give unexpected results?

**Comment by bobwhitelock** [comment]
I was incorrect, this is apparently because @WilliamMcCumstie has wrapped false so it is now true...

**Comment by WilliamMcCumstie** [comment]
It was somewhat a regression to `2017.1` issue. Previously `'false'` was manually converted to `false` and this solved the issue. However changing the rendering process means the conversion occurs before the wrapper and thus `false` had to be handled.

**Comment by WilliamMcCumstie** [comment]
The `NilDetectionWrapper` now contains a check if the return value is falsy. It no longer wraps any falsy values. The change has been made on the `release/2017.2.0` branch which will be merged into `develop` once the release is complete.
